sw/host/rom_ext_image_tools/signer/image/src/image.rs

```diff
@@ -91,6 +136,49 @@ impl RawImage {
 
         fs::write(output_file, &self.data).expect("Failed to write the new binary file!");
     }
+
+    /// Updates ROM_EXT manifest usage constraints field.
+    fn update_usage_constraints_field(&mut self, path: &Path) {
+        // Update fields from config.
+        let usage_constraints = fs::read(path).expect("Failed to read usage constraints!");
+
+        let offset = manifest::ROM_EXT_USAGE_CONSTRAINTS_OFFSET;
+        self.update_field(&usage_constraints, offset);
+    }
+
+    /// Updates ROM_EXT manifest peripheral lockdown info field.
+    ///
+    /// The information is encoded into the 128-bit binary blob.
+    fn update_peripheral_lockdown_info_field(&mut self, _info: &PeripheralLockdownInfo) {
+        // TODO: generate the peripheral_lockdown_blob from
+        //       PeripheralLockdownInfo, meanwhile use a hard-coded vector.
+
+        let offset = manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO_OFFSET;
+        self.update_field(&[0xA5; 16], offset);
+    }
+
+    /// Updates ROM_EXT manifest timestamp field.
+    ///
+    /// The generated time stamp is u64. Normally time stamp is a signed
+    /// integer, however there is no risk of overflow into a "negative".
+    fn update_timestamp_field(&mut self) {
+        let duration = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Failed to obtain the current time!");
+
+        let offset = manifest::ROM_EXT_IMAGE_TIMESTAMP_OFFSET;
```

Comment:
In general I'm against single-use variables.

Reply:
I have fixed all the places in `image.rs` where single use variables simply alias a field or similar.